src/app/(main)/layout.tsx

```diff
@@ -17,14 +19,30 @@ export default function LearnerLayout({
         router.push('/');
     }
 
+    const useAdminStatus = () => {
+        const [isAdmin, setIsAdmin] = useState(false);
+        useEffect(() => {
+            const getUserProfile = callApi('getUserProfile');
```

Comment:
Can use auth.currentUser.getIdTokenResult() (auth object from firebase file below)

Reply:
You were right, that's a much better way to do it. Thank you!